app/src/main/java/org/listenbrainz/android/data/repository/YimRepository.kt

```diff
@@ -5,4 +5,8 @@ import org.listenbrainz.android.util.Resource
 
 interface YimRepository {
     suspend fun getYimData(username: String): Resource<YimData>
+    
+    fun getUsername(): String?
```

Comment:
This and the login status shouldn't be a part of this repository. Please remove these from here and pass these as parameters to the functions.

Reply:
The tests again don't work as these preferences need app context and thats un-mockable